### PR TITLE
fix: added ability to hash password

### LINKED ISSUE #17: [Fix]: Hashing passwords before saving in the DB
## Description
In the current state, we are saving the passwords directly in the database without having them. Which is not the best practice. 

## Solution
We can make use of Bcrypt to hash the password and then save it in the database and we can decrypt the passwords using another Bcypt function. 

## Context
image

I would like to work on this issue!

### PR DESCRIPTION
**Pull request by SanjaySinghRajpoot**
## Description
- Added ability to hash passwords using the bcrypt library 
- Added a new util which contains two functions `HashPassword` and `VerifyPassword` 
- Hashed the password in the Create User function

## Context
Fixes #17

### CODE REVIEW

**Review by championswimmer — CHANGES_REQUESTED** [review]
Please add unit tests for hash and verify password

**Comment by SanjaySinghRajpoot** [comment]
> Please add unit tests for hash and verify password

I have added them you can check now.

File: src/controllers/users.go [hunk 1]
```diff
@@ -19,9 +20,16 @@ func NewUsersController(db *gorm.DB) *UsersController {
 
 // Create new user
 func (c *UsersController) Create(email string, password string) (*models.User, error) {
+
+	hashedPassword, err := utils.HashPassword(password)
+
+	if err != nil {
```
Comment (championswimmer) [inline src/controllers/users.go:26]: this call should not fail (under any normal operation)
Can you do a `lo.Must` wrapper simply instead ?

File: src/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"fmt"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+func HashPassword(password string) (string, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+
+	if err != nil {
+		return "", fmt.Errorf("could not hash password %w", err)
```
Comment (championswimmer) [inline src/utils/utils.go:13]: this call should not fail (under any normal operation)
Can you do a `lo.Must` wrapper simply instead ?

**Review by championswimmer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**